profiles/audit_log.py

```diff
@@ -123,83 +124,42 @@ def log(action, instance):
 
-        for (action, profile_part), timestamp in data["parts"].items():
-            log_entries.append(
-                LogEntry(
-                    timestamp=timestamp,
-                    service_name=service_name,
-                    client_id=client_id,
-                    ip_address=ip_address,
-                    actor_user_id=actor_user_id,
-                    actor_role=actor_role,
-                    target_user_id=target_user_uuid,
-                    target_profile_id=profile_id,
-                    target_type=profile_part,
-                    operation=action,
-                )
+        for action, profile_part in data["parts"]:
+            entry = StructuredResilientLogEntryData(
+                level=logging.NOTSET,
+                message=status,
+                actor={
+                    "user_id": str(actor_user_id) if actor_user_id else None,
+                    "role": actor_role,
+                    "ip_address": ip_address,
+                    "client_id": client_id,
+                    "service": service_name,
+                },
+                operation=action,
+                target={
+                    "user_id": str(target_user_uuid),
+                    "profile_id": str(profile_id),
+                    "type": profile_part,
+                },
+                extra={"status": status},
             )
-
-    return log_entries
-
-
-def _put_logs_to_logger(log_entries):
-    logger = logging.getLogger("audit")
-
-    for log_entry in log_entries:
-        actor_dict = {
-            k: v
-            for k, v in [
-                ("service_name", log_entry.service_name),
-                ("client_id", log_entry.client_id),
-                ("ip_address", log_entry.ip_address),
-                ("user_id", str(log_entry.actor_user_id or "")),
-                ("role", log_entry.actor_role),
-            ]
-            if v
-        }
-
-        target_dict = {
-            k: v
-            for k, v in [
-                ("id", str(log_entry.target_profile_id or "")),
-                ("user_id", str(log_entry.target_user_id or "")),
-                ("type", log_entry.target_type),
-            ]
-            if v
-        }
-
-        message = {
-            "audit_event": {
-                "origin": "PROFILE-BE",
-                "status": "SUCCESS",
-                "date_time_epoch": int(log_entry.timestamp.timestamp() * 1000),
-                "date_time": f"{log_entry.timestamp.replace(tzinfo=None).isoformat(sep='T', timespec='milliseconds')}Z",  # noqa: E501
-                "actor": actor_dict,
-                "operation": log_entry.operation,
-                "target": target_dict,
-            }
-        }
-
-        logger.info(json.dumps(message))
-
-
-def _put_logs_to_db(log_entries):
-    LogEntry.objects.bulk_create(log_entries)
+            entries.append(entry)
+        ResilientLogSource.bulk_create_structured(entries)
```

Comment:
Wrong indentation here or missing a reset of `entries` in the outer loop?


```suggestion
    ResilientLogSource.bulk_create_structured(entries)
```